### DISCUSSION TITLE
How to route Cardinal midi to another midi in Ableton Live?

### DISCUSSION THREAD

**Discussion by irish-moss** [DISTRHO/Cardinal#299] [Q&A]
I recently just watched the great Venus Theory youtube video on Cardinal and followed his beginner instructions word for word. I had good luck with the first part after a few errors. The second part of the video dealt with using cardinal as a sequencer generator to control another synth. I tried to do this on Ableton but I cannot get it to link. I see the cardinal track but on the second combi box there is only pre and post and not a selection for cardinal synth. I am confused what I am doing wrong. Any help would be appreciated. I'm making progress but I can't figure this one out.

**Comment by dromer** [comment]
Have you ever used other MIDI plugins in Ableton?
And which plugin version are you using? (VST2, VST3 and which variant)

**Reply by irish-moss** [reply]
Thanks for reply. I'm using the AU Cardinal, I believe newest one. I have used ableton for apx 16 years and yes much midi.

**Reply by dromer** [reply]
FX or Synth? Maybe try with the VST2 instead?

**Reply by irish-moss** [reply]
Ah! I believe this is who I spoke to on Venus' video and you said you weren't Ableton equipped.

**Reply by dromer** [reply]
Indeed I do not have a machine with Ableton, but I can try and get more information out of you since all of this is missing from your report.

Anyone following along will need to know enough details to either reproduce or come with suggestions on what you can try to resolve the setup.

**Reply by irish-moss** [reply]
Much appreciated. I've never had an issue with routing midi to another midi track. I know there is some simple thing I am not comprehending. Seems Cameron just dropped the next synth after Cardinal and was fine in Bitwig.

**Reply by irish-moss** [reply]
A youtube follower of Venus Theory commented this 

open a blank "cardinal sin of the EDM" instance,  then locate CV*MIDI module, Whatever midi you want to send, you gotta send voltages/triggers to both the v/oct and Gate outs of that module.   By default in that CV*MIDI module there must be already selected "PLUGINDRIVER" as Midi driver, "Cardinal" as device.  Now in Ableton:  create a new midi track, choose the cardinal track in the "Ins" combobox e.g ( "1-CardinalSynth"), next combobox pick  "Cardinalsynth", change the monitor button of that track to In (In/Auto/off)  and there you go , there must be midi routed back to Ableton.


I am confused if by cv midi he means the midi host that opens with new project.

**Reply by dromer** [reply]
Maybe give an example patch of what you are trying.

However you still didn't answer: which plugin variant are you using (FX or Synth) and if you have tried VST2 (or VST3 perhaps, even though this is Beta).

**Reply by irish-moss** [reply]
I'm using synth. I was able to build a basic reverb in fx and send audio through it. I will try vst3 if I still have it. I generally lean towards AU for stability

The patch I tried to monitor was exactly as Cameron/Venus Theory did in the second part of his tutorial, yesterday. A basic sequencer for generative.

**Reply by falkTX** [reply]
try using the FX variant. Ableton might be ignoring MIDI output from plugins with an "instrument" tag, such as the case for the cardinal synth variant

**Reply by dromer** [reply]
VST3 is beta. AU is somewhat experimental/unique, it's a custom wrapper. Most stable for you should be VST2.

Please try all plugin formats and variants. If any (or all) don't work with midi in Ableton this needs deeper investigation of course.

**Reply by falkTX** [reply]
all of them should work at this point. with everything supporting at least MIDI notes

**Reply by irish-moss** [reply]
Here's where I'm at. I followed Dromer's and your advice. I downloaded the pkg again and installed. Rescanned. Ableton sees Cardinal synth vst2 and vst3 in the second combi box like other midi synths on ableton. Problem is, neither will open any presets. I hit open/import and just resets back to the dropdown menu. :/

**Reply by irish-moss** [reply]
And now I apparently can't open preset or save on AU, VST and VST3 after reinstalling. :( 
I tried a log out of mac with no improvement. Guess I'll try a restart. This is gonna hurt if it doesn't correct.

**Reply by irish-moss** [reply]
I've rebooted the machine. Drop down selection in all AU, vst and vst3 won't allow open. Command + O will. But all of the presets I have worked at so far and saved are grayed out and cannot be selected. :/

**Reply by irish-moss** [reply]
Nope. Command + O is opening for Ableton Live and not within the VST. So....I think I'm temporarily done with Cardinal right as I've got my legs. Wishing I hadn't reinstalled the AU today with the VST and VST3. 

Any idea why I can't open a preset with the drop  down menu and mouse?

**Reply by falkTX** [reply]
the 22.06 macos-intel pkg has that bug, use the macos-universal pkg. this will be fixed in 22.07 in 3 days

**Reply by irish-moss** [reply]
Ok thank you.

**Reply by irish-moss** [reply]
Reinstalled Universal. Open and Save functioning. 
Opened VST preset I did with sequencer. Set up monitor to vital. Nothing. Started playing with the midi host. By patching my ADSR env out to midi host gate, Vital began to play. 
And that last part is illustrative of old fashioned ignorance on my part and is probably what the youtube commenter was trying to show me in post above. Well....Improvement! LOL
Thank you for your time and guidance.

**Comment by falkTX** [comment]
for such cases, I think it makes sense to create a Host MIDI monitor module. just to get debug information in regards to what messages are coming in, and what is going out.

**Reply by irish-moss** [reply]
Are you saying like a secondary midi host that monitors the output of the primary?

**Reply by falkTX** [reply]
just a panel with text information, a scrolling list of events

**Reply by irish-moss** [reply]
interesting

**Reply by irish-moss** [reply]
Again, modular is new territory for me.
